src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/CreateTableRequestFactory.java

```diff
@@ -21,14 +21,26 @@
  */
 public interface CreateTableRequestFactory {
 
-    /*
-     * Takes a virtual table description and returns a CreateTableRequest corresponding physical table if one is found.
+    /**
+     * Takes a virtual table description and returns a CreateTableRequest for the corresponding static physical table
+     * if one is found.
      */
-    Optional<CreateTableRequest> getCreateTableRequest(DynamoTableDescription virtualTableDescription);
+    Optional<CreateTableRequest> getStaticPhysicalTable(DynamoTableDescription virtualTableDescription);
 
-    /*
-     * Returns a list of CreateTableRequests that will be created when the factory is initialized.
+    /**
+     * Returns a list of CreateTableRequests for the static physical tables created when the factory is initialized.
      */
-    List<CreateTableRequest> getPhysicalTables();
+    List<CreateTableRequest> getStaticPhysicalTables();
+
+    /**
+     * Takes a virtual table description and returns a CreateTableRequest for the corresponding physical table to be
+     * created dynamically.
+     */
+    CreateTableRequest getDynamicPhysicalTable(DynamoTableDescription virtualTableDescription);
```

Comment:
This tells us how to create a new physical table on the fly when there's a request to create a new multitenant table